packages/medusa/src/api/routes/admin/variants/get-inventory.ts

```diff
@@ -118,9 +123,18 @@ export default async (req, res) => {
           }
         }
 
-        const quantity = await inventoryService.retrieveAvailableQuantity(
-          inventory[0].id,
-          channel.locations
+        const quantity = Math.min(
+          ...(await Promise.all(
+            variantInventoryItems.map(async (variantInventory) => {
+              return (
+                // eslint-disable-next-line max-len
+                (await salesChannelInventoryService.retrieveAvailableItemQuantity(
+                  channel.id,
+                  variantInventory.inventory_item_id
+                )) / variantInventory.required_quantity
+              )
+            })
+          ))
```

Comment:
thought: now that we have this in two places should we consider extracting it in a util - or maybe add it as a method in the ProductVariantInventoryService?